src/routeGeneration/templateHelpers.ts

```diff
@@ -359,6 +363,95 @@ export class ValidationService {
     return new Buffer(value);
   }
 
+  public validateUnion(name: string, value: any, fieldErrors: FieldErrors, subSchemas: TsoaRoute.PropertySchema[] | undefined, parent = ''): any {
+    if (!subSchemas) { return; }
+
+    const subFieldErrors = subSchemas.map(subSchema => {
+      const subFieldErrors: FieldErrors = {};
+      this.ValidateParam(subSchema, value, name, subFieldErrors, parent);
+      return subFieldErrors;
+    });
+
+    if (!subFieldErrors.some(subFieldError => Object.keys(subFieldError).length === 0)) {
+      fieldErrors[parent + name] = {
+        message: `Could not match the union against any of the items. Issues: ${subFieldErrors}`,
+        value,
+      };
+      return;
+    }
+
+    return value;
+  }
+
+  public validateIntersection(name: string, value: any, fieldErrors: FieldErrors, subSchemas: TsoaRoute.PropertySchema[] | undefined, parent = ''): any {
+    if (!subSchemas) { return; }
+
+    const subFieldErrors = subSchemas.map(subSchema => {
+      const subFieldErrors: FieldErrors = {};
+      this.ValidateParam(subSchema, value, name, subFieldErrors, parent);
+      return subFieldErrors;
+    });
+
+    subFieldErrors.forEach(subFieldError => {
+      (Object as any).entries(subFieldError).forEach(([key, value]) => {
```

Comment:
I think we can't use `.entries` out of the box unless TypeScript is transpiling it down to a version that Node 6 supports ( https://node.green/#ES2017-features-Object-static-methods-Object-entries ). I realize that it's probably ridiculous to support Node 6 at this point, but to be safe, let's add `"es2017"` to the `"lib"` section of tsconfig.json.